packages/visualizations/src/components/MapPoi/Map.ts

```diff
@@ -259,13 +303,14 @@ export default class MapPOI {
         // - Selected feature is the same as the active feature: This means that I clicked on the feature for which the popup is open.
         if (!hasFeatures || isSelectedFeatureSameAsActiveFeature) {
             this.popup.remove();
+            this.availableFeaturesOnClick = [];
             return;
         }
 
         // FIXME: remove eslint comment.
         // eslint-disable-next-line prefer-destructuring
         this.activeFeature = features[0];
-
+        this.availableFeaturesOnClick = features;
```

Comment:
As popup is an optional configuration for a layer we must filter `availableFeaturesOnClick` with only feature that has a popup config.

Reply:
Done here https://github.com/opendatasoft/ods-dataviz-sdk/pull/209/commits/7531261c7c401d9a3b2b90bcd2462ec342dd80d6